packages/core-mobile/app/screens/rpc/components/shared/DetailSectionView.tsx

```diff
@@ -0,0 +1,117 @@
+import { getDateInMmmDdYyyyHhMmA } from 'utils/date/getDateInMmmDdYyyyHhMmA'
+
+export const DetailSectionView = ({
+  detailSection,
+  onPressDataItem
+}: {
+  detailSection: DetailSection
+  onPressDataItem: (data: string) => void
+}): JSX.Element => {
+  const avaxPrice = useSelector(selectAvaxPrice)
+  const selectedCurrency = useSelector(selectSelectedCurrency)
+  const { tokenInCurrencyFormatter } = useApplicationContext().appHook
+
+  return (
+    <View>
+      {detailSection.title && (
+        <View sx={{ flexDirection: 'row', justifyContent: 'space-between' }}>
+          <Text variant="buttonMedium">{detailSection.title}</Text>
+        </View>
+      )}
+      <View
+        sx={{
+          justifyContent: 'space-between',
+          marginTop: 12,
+          borderRadius: 8,
+          padding: 12,
+          marginBottom: 12,
+          gap: 8,
+          backgroundColor: '$neutral800'
+        }}>
+        {detailSection.items.map((item, index) => {
+          if (typeof item === 'string') {
+            return (
+              <View key={index}>
+                <Text variant="body1">{item}</Text>
+              </View>
+            )
+          } else if (item.type === 'text') {
+            return (
+              <Row
+                style={
+                  item.alignment === 'horizontal'
+                    ? {
+                        justifyContent: 'space-between',
+                        alignItems: 'center'
+                      }
+                    : {
+                        flexDirection: 'column'
+                      }
+                }
+                key={index}>
+                <Text variant="caption">{item.label}</Text>
+                <Text variant="buttonSmall">{item.value}</Text>
+              </Row>
+            )
+          }
+
+          return (
+            <Row
+              style={{
+                justifyContent: 'space-between',
+                alignItems: item.type === 'currency' ? 'flex-start' : 'center'
+              }}
+              key={index}>
+              <Text variant="caption">{item.label}</Text>
+              {item.type === 'address' ? (
+                <TokenAddress address={item.value} />
+              ) : item.type === 'nodeID' ? (
+                <NodeID nodeID={item.value} />
+              ) : item.type === 'data' ? (
+                <TouchableOpacity
+                  hitSlop={12}
+                  onPress={() => onPressDataItem(item.value)}>
+                  <Text variant="buttonSmall" sx={{ color: '$blueDark' }}>
+                    View
+                  </Text>
+                </TouchableOpacity>
+              ) : item.type === 'date' ? (
+                <Text variant="buttonSmall">
+                  {getDateInMmmDdYyyyHhMmA(parseInt(item.value))}
+                </Text>
+              ) : (
+                <View sx={{ alignItems: 'flex-end' }}>
+                  <Text variant="buttonSmall">
+                    {new TokenUnit(
```

Comment:
can you extract this block into either small functions or components? will be easier to follow

Reply:
added rendering functions